app/controllers/posts_controller.rb

```diff
@@ -17,13 +18,21 @@ def create
     @user = current_user
     @post = @user.posts.new(post_params)
     if @post.save
-      redirect_to user_post_path(@user, @post)
+      redirect_to user_posts_path(@user)
     else
       flash.now[:errors] = 'Invalid post!'
       render :new
     end
   end
```

Comment:
- [ ] Please `CanCanCan` for Post authorization in `Post Controller`
add `load_and_authorize_resource`